Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -75,6 +80,40 @@ public FreezeOnMatchCustomization(Type targetType)
         public IRequestSpecification Matcher { get; }
 
         /// <summary>
+        /// Used to create requests for resolving specimens before freezing.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// By default, the factory returns TargetType, but it can be tweaked 
+        /// to return any kind of request.
+        /// </para>
+        /// <para>
+        /// A typical use case is to create a request which includes significant 
+        /// information about a specimen. For instance, <see cref="ParameterInfo"/> 
+        /// which is widely used in Glue Libraries can contain custom attributes
+        /// such as <see cref="StringLengthAttribute"/> or <see cref="RegularExpressionAttribute" />.
+        /// These attributes are essential part of the request since they influence 
+        /// the final specimen.
+        /// </para>
+        /// </remarks>
+        public Func<object> RequestFactory
```

Comment:
That's a good question. I started with a virtual `CreateRequest` method that could create a _proper_ request in Glue Libraries (overriding the default implementation in new classes like `xUnit2.FreezeOnPatameterMatchCustomization`). Then I realized that the method isn't really necessary and we can easily avoid creating such classes using delegates...

I haven't really considered using a property (maybe because we've [discussed it before](https://github.com/AutoFixture/AutoFixture/pull/665#discussion_r69581627))... It seems to me that the factory method here is a bit more eloquent. Please compare how it might be used in the `FrozenAttribute`:

```csharp
return new FreezeOnMatchCustomization(type, filter) {
    RequestFactory = () => parameter
};
```

```csharp
return new FreezeOnMatchCustomization(type, filter) {
    Request = parameter
};
```